### PR TITLE
feat(meetings): merge beta into next

### PR DESCRIPTION
**Pull request by meta-dipanshu-sharma**
# COMPLETES #< [SPARK-449062](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-449062) >

## This pull request addresses

We need to merge beta into next branch for release of SDK V3.

## by making the following changes

Created a PR targeted against next branch from beta branch

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,14 +26,14 @@
...
   },
   "scripts": {
-    "clean": "rimraf packages/**/dist packages/**/types",
+    "clean": "rimraf \"packages/**/.coverage\" \"packages/**/dist\" \"packages/**/types\"",
     "prebuild": "yarn run clean",
     "build:publish": "yarn build:package && yarn workspaces foreach --parallel --verbose run build:src",
     "build:local": "yarn workspaces foreach --parallel --verbose run build:src && yarn run samples:build",
     "build": "yarn workspace @webex/calling run build:src && node ./tooling/index.js build && yarn && yarn ws:tools build && yarn run build:tsc",
-    "build:tsc": "yarn workspace @webex/plugin-meetings run build && yarn run tsc -p ./config/tsconfig.typecheck.json --noEmit",
+    "build:tsc": "yarn run tsc -p ./config/tsconfig.typecheck.json --noEmit && yarn workspace @webex/internal-plugin-metrics run build && yarn workspace @webex/plugin-meetings run build && yarn workspace @webex/media-helpers run build:src",
```
Comment (meta-dipanshu-sharma) [inline package.json:34]: build `media-helpers` before `plugin-meetings`

Comment (sreenara) [inline package.json:34]: The media-helpers package should be built first followed by plugin-meetings IMO. Observed issues with calling in the `next` branch.

Comment (sreenara) [inline package.json:34]: We need a new ticket to simplify the build commands. Having these many build commands can cause confusion

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -24,6 +24,7 @@ This is for compliance purposes with FedRAMP program.
 ### Change Type
 
 <!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
+
```
Comment (sreenara) [inline .github/PULL_REQUEST_TEMPLATE.md:27]: Is this change required?

File: package.json [hunk 3]
```diff
@@ -17,7 +17,7 @@
   "license": "Cisco EULA (https://www.cisco.com/c/en/us/products/end-user-license-agreement.html)",
   "author": "devsupport@webex.com",
   "engines": {
-    "node": ">=14.x <=16.x",
+    "node": "16.x",
```
Comment (meta-dipanshu-sharma) [inline package.json:20]: Should we make it `>=16.x`

File: CHANGELOG.md [hunk 4]
```diff
@@ -373,57 +373,350 @@ All notable changes to this project will be documented in this file. See [standa
 
 * **plugin-conversation:** Send kms on join ([#2568](https://github.com/webex/webex-js-sdk/issues/2568)) ([0960a66](https://github.com/webex/webex-js-sdk/commit/0960a66830559d01656b04cfc0642e7ff820e041))
 
-### [2.37.1](https://github.com/webex/webex-js-sdk/compare/v2.37.0...v2.37.1) (2023-01-17)
+### [2.37.1](https://github.com/webex/webex-js-sdk/compare/v3.0.0-beta.14...v2.37.1) (2023-01-17)
```
Comment (sreenara) [inline CHANGELOG.md:376]: What do we want to do about the CHANGELOG? Which changes do we want to keep?

Comment (meta-dipanshu-sharma) [inline CHANGELOG.md:376]: Should we remove these beta changelogs?

File: .vscode/settings.json [hunk 5]
```diff
@@ -22,7 +22,7 @@
   },
   "editor.defaultFormatter": "esbenp.prettier-vscode",
   "eslint.format.enable": true,
-  "breadcrumbs.enabled": false,
+  "breadcrumbs.enabled": true,
```
Comment (sreenara) [inline .vscode/settings.json:25]: Is this required?

File: tooling/modified-packages.js [hunk 6]
```diff
@@ -38,11 +38,11 @@ const fileToPackage = (d) => {
  * @param {boolean=} options.dependents if true, also includes dependents of
  * updated packages
  * @param {boolean=} options.npm if true, compares to the last tag published to
- * npm instead of upstream/master
+ * npm instead of upstream/beta
```
Comment (sreenara) [inline tooling/modified-packages.js:41]: What is happening here? Should we change this to `next`?

File: tooling/lib/version.js [hunk 7]
```diff
@@ -197,12 +197,12 @@ async function checkLastCommit() {
 }
 
 /**
- * Determines if there are any breaking changes between HEAD and upstream/master
+ * Determines if there are any breaking changes between HEAD and upstream/beta
```
Comment (sreenara) [inline tooling/lib/version.js:200]: Should we change `beta` to `next`

File: tooling/lib/updated.js [hunk 8]
```diff
@@ -15,11 +15,11 @@ const {diff} = require('./git');
  * @param {boolean=} options.dependents if true, also includes dependents of
  * updated packages
  * @param {boolean=} options.npm if true, compares to the last tag published to
- * npm instead of upstream/master
+ * npm instead of upstream/beta
  * @returns {Promise<Array<string>>}
  */
 exports.updated = async function updated({dependents, npm = !!process.env.CI}) {
-  const tag = npm ? await last() : 'upstream/master';
+  const tag = npm ? await last() : 'upstream/beta';
```
Comment (sreenara) [inline tooling/lib/updated.js:22]: Should we change this to `next`?

**Review by sreenara — CHANGES_REQUESTED** [review]
Please see the comments

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (meta-dipanshu-sharma) [reply .github/PULL_REQUEST_TEMPLATE.md:27]: It was added in the beta version so was added in this PR as well. Should we remove it?

File: package.json (hunk 3 above)
Comment (torgeadelin) [reply package.json:20]: 16 is end of life as of September 11, we should probabbly try a newer version? https://nodejs.org/en/blog/announcements/nodejs16-eol

File: .vscode/settings.json (hunk 5 above)
Comment (torgeadelin) [reply .vscode/settings.json:25]: yes please, i would like for this to be merged over too, it's very annoying not to have the breadcrumbs when u have opened files in vs code

File: tooling/lib/version.js (hunk 7 above)
Comment (mkesavan13) [reply tooling/lib/version.js:200]: I think this isn't effective in the next branch. It was last updated in `next` branch 10 months ago and it still says `upstream/master`

So, I believe it is not relevant anymore

File: tooling/modified-packages.js (hunk 6 above)
Comment (mkesavan13) [reply tooling/modified-packages.js:41]: I think this isn't effective in the next branch. It was last updated in `next` branch 10 months ago and it still says `upstream/master`

So, I believe it is not relevant anymore

CC - @InteractiveTimmy

File: CHANGELOG.md (hunk 4 above)
Comment (mkesavan13) [reply CHANGELOG.md:1629]: I think we need not worry about the change log as we have the change logs for the respective releases in their respective branches. I remember that's what Marcin mentioned in the space as well. As long as we keep beta and master branches, changelog shouldn't matter
